extensions/ql-vscode/src/remote-queries/variant-analysis-manager.ts

```diff
@@ -103,6 +114,14 @@ export class VariantAnalysisManager extends DisposableObject implements VariantA
     return this.variantAnalyses.get(variantAnalysisId);
   }
 
+  public getVariantAnalyses(): Map<number, VariantAnalysis> {
```

Comment:
This is quite risky/leaky. It gives any part of the codebase access to get and edit variant analyses. If we _have to_ expose them, we should try and protect against random changes to the data by returning a copy of the real data. Perhaps using [immutablejs](https://immutable-js.com/) which is already used in the codebase.

Reply:
I've removed this public method and replaced it with a smaller one which just returns the size: https://github.com/github/vscode-codeql/pull/1655/commits/51b547b411d8afd09eaa1f3f87d18c43b61970d7